### ISSUE TITLE
drop tui support

### ISSUE THREAD

**Issue by pm100** [rhysd/tui-textarea#30]
since tui is now basically dead, textarea should drop support for it and only support ratatui. Why? because new features are being added to ratatui and textarea cannot take advantage of them without getting very messy.

 See https://github.com/rhysd/tui-textarea/pull/29

**Comment by joshka** [comment]
As I understand it, @rhysd's plan is to make `ratatui` the default in the next release and then deprecate tui in a release after that.

**Comment by rhysd** [comment]
Yeah, I don't want to remove compatibility as much as possible until there is some strong motivation.

tui.rs was going inactive, but it is still usable and actually many apps are depending on it.

**Comment by pm100** [comment]
OK I will resumbit #29 using the ratatui masked feature. Please can you look at the PR sooner rather than later (once I have resubmitted it) . It took 5 months to get my last request through and my work on gitui was blocked for that duration.

**Comment by rhysd** [comment]
@pm100 Yes, that's great. Please ping me by @rhysd mention when my response is not good.

**Comment by pm100** [comment]
@rhysd - ok the PR is there now. But I had a change of plan. The actual mask code was fairly simple - in fact just a one liner. But it got very complicated with all the feature combinations and fiddly due to it having to be in several places. So instead I just lifted the one liner from ratatuis mask code and dropped it in to replace_tabs, which is really 'prepare_line_for_display'. So now it works with tui and ratatui

**Comment by rhysd** [comment]
Let me close this for now since there is no strong motivation to do this.
